Bottom: a2de1caa8d89582626ed66f6fbd39df185a3a3b1
Top:    5726e917e70c8c52eb28a24c3d960654443a61dc
Author: Jason Wang <jasowang@redhat.com>
Date:   2016-11-08 15:46:28 +0800

Refresh of vhost_net-tx-support-batching

---

diff --git a/drivers/vhost/net.c b/drivers/vhost/net.c
index 87c745c..3308440 100644
--- a/drivers/vhost/net.c
+++ b/drivers/vhost/net.c
@@ -461,11 +461,12 @@ static void handle_tx(struct vhost_net *net)
 		total_len += len;
 		if (vq->delayed < tx_batched &&
 		    total_len < VHOST_NET_WEIGHT &&
-		    !vhost_vq_avail_empty(&net->dev, vq)) &&
+		    !vhost_vq_avail_empty(&net->dev, vq)) {
 			vq->delayed++;
 			msg.msg_flags |= MSG_MORE;
 		} else {
 			vq->delayed = 0;
+			msg.msg_flags &= ~MSG_MORE;
 		}
 		/* TODO: Check specific error and bomb out unless ENOBUFS? */
 		err = sock->ops->sendmsg(sock, &msg, len);
